contracts/DCNT721A.sol

```diff
@@ -96,13 +127,81 @@ contract DCNT721A is ERC721A, Initializable, Ownable, Splits {
     }
 
     _safeMint(msg.sender, numberOfTokens);
-    for (uint256 i = 0; i < numberOfTokens; i++) {
-      emit Minted(msg.sender, mintIndex++);
+    unchecked {
+      for (uint256 i = 0; i < numberOfTokens; i++) {
+        emit Minted(msg.sender, mintIndex++);
+      }
+    }
+  }
+
+  // allows the owner to "airdrop" users an NFT
+  function mintAirdrop(address[] calldata recipients) external onlyOwner {
+    uint256 atId = _nextTokenId();
+    uint256 startAt = atId;
+    require(atId + recipients.length <= MAX_TOKENS,
+      "Purchase would exceed max supply"
+    );
+
+    unchecked {
+      for (
+        uint256 endAt = atId + recipients.length;
+        atId < endAt;
+        atId++
+      ) {
+        _safeMint(recipients[atId - startAt], 1);
+        emit Minted(recipients[atId - startAt], atId);
+      }
+    }
+  }
+
+  function mintPresale(
+    uint256 quantity,
+    uint256 maxQuantity,
+    uint256 pricePerToken,
+    bytes32[] calldata merkleProof
+  )
+    external
+    payable
+    verifyTokenGate(true)
+  {
+    require (block.timestamp >= presaleStart && block.timestamp <= presaleEnd, 'not presale');
+    uint256 mintIndex = _nextTokenId();
+    require(!saleIsPaused, "Sale must be active to mint");
+    require(
+      mintIndex + quantity <= MAX_TOKENS,
+      "Purchase would exceed max supply"
+    );
+    require (MerkleProof.verify(
+        merkleProof,
+        presaleMerkleRoot,
+        keccak256(
+          // address, uint256, uint256
+          abi.encode(msg.sender,maxQuantity,pricePerToken)
+        )
+      ), 'not approved');
+
+    require(msg.value >= (pricePerToken * quantity), "Insufficient funds");
+    require(balanceOf(msg.sender) + quantity <= maxQuantity, 'minted too many');
+    _safeMint(msg.sender, quantity);
+    unchecked {
+      for (uint256 i = 0; i < quantity; i++) {
+        emit Minted(msg.sender, mintIndex++);
+      }
     }
   }
 
   function flipSaleState() external onlyOwner {
```

Comment:
```suggestion
  /// @notice pause or unpause sale 
  function flipSaleState() external onlyOwner {
```

Reply:
OpenZeppelin also has a `Pausable` contract. Have we considered using this instead of our own methods?

https://docs.openzeppelin.com/contracts/2.x/api/lifecycle#pausable